File: src/java/org/javaturk/dp/ch05/adapter/electricity/domain/us/USPowerProvider.java
package org.javaturk.dp.ch05.adapter.electricity.domain.us;

public class USPowerProvider implements USPowerSource {
	private boolean on;

	public void providePowerAt110V() {
		System.out.println("USPowerProvider provides electricity at 110V!");
	}

	public USPowerProvider() {
		System.out.println("USPowerProvider is up and running.");
	}

	public void pushSwitch() {
		if (!on) {
			on = true;
			System.out.println("USPowerProvider started to provide electricity.");
		} else {
			on = false;
			System.out.println("USPowerProvider stopped to provide electricity.");
		}
	}
}

File: src/typescript/com/gokselkucuksahin/dp/structural-patterns/adapter/electricity/domain/us/us-power-provider.ts
import {USPowerSource} from "./us-power-source";

export class USPowerProvider implements USPowerSource {
  private on: boolean;

  constructor() {
    console.log("USPowerProvider is up and running.");
  }

  providePowerAt110V(): void {
    console.log("USPowerProvider provides electricity at 110V!");
  }

  pushSwitch(): void {
    if (this.on) {
      console.log("USPowerProvider stopped to provide electricity.");
    } else {
      console.log("USPowerProvider started to provide electricity.");
    }
    this.on = !this.on;
  }
}